### ISSUE TITLE
NStepLSTM does not work with updater, evaluater

### ISSUE THREAD

**Issue by ywatanabex** [chainer/chainer#1712]
I found that NStepLSTM does not work with updater and evaluator.

This is because current updater and evaluator implementations assume that each argument of a link call is `chainer.Variable`.  For most of the links, this is true. But `NStepLSTM` is an exception; its argument `xs` is a `list of chainer.Variable`.

A quick fix for this is to use the following rule:
- if array is given, convert it to chainer.Variable  (as before)
- if list is given, convert each element to chainer.Variable (new rule)

```
# chainer/training/updater.py
in_arrays = self.converter(batch, self.device)
if isinstance(in_arrays, tuple):
    in_vars = tuple(variable.Variable(x, volatile='on') for x in in_arrays)   # this is current code
```

Does this make sense?
Are there better way to solve this problem?

**Comment by unnonouno** [comment]
Thank you for reporting the problem. Sorry i know this problem, and we now trying to remove Variable wrap itself: #1654. Actually, Function automatically wraps an ndarray with Variable: https://github.com/pfnet/chainer/blob/master/chainer/function.py#L180
Note that n_step_lstm doesn't support ndarray now. #1703

**Comment by ywatanabex** [comment]
Thank you. I understand the direction.
I'm looking forward to updates.

**Comment by unnonouno** [comment]
This problem is solved via #1654.
